// plugin-welcome/src/api/models/ControllersPatient.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersPatient
 */
export interface ControllersPatient {
    /**
     * 
     * @type {number}
     * @memberof ControllersPatient
     */
    gender?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersPatient
     */
    hn?: string;
    /**
     * 
     * @type {string}
     * @memberof ControllersPatient
     */
    patientName?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersPatient
     */
    rightoftreatment?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersPatient
     */
    systemmember?: number;
}

export function ControllersPatientFromJSON(json: any): ControllersPatient {
    return ControllersPatientFromJSONTyped(json, false);
}

export function ControllersPatientFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersPatient {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'gender': !exists(json, 'gender') ? undefined : json['gender'],
        'hn': !exists(json, 'hn') ? undefined : json['hn'],
        'patientName': !exists(json, 'patientName') ? undefined : json['patientName'],
        'rightoftreatment': !exists(json, 'rightoftreatment') ? undefined : json['rightoftreatment'],
        'systemmember': !exists(json, 'systemmember') ? undefined : json['systemmember'],
    };
}

export function ControllersPatientToJSON(value?: ControllersPatient | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'gender': value.gender,
        'hn': value.hn,
        'patientName': value.patientName,
        'rightoftreatment': value.rightoftreatment,
        'systemmember': value.systemmember,
    };
}
